### PR TITLE
[11.0] [mig] printer_zpl2

### PR DESCRIPTION
**Pull request by bodedra**

### CODE REVIEW

File: printer_zpl2/tests/test_printing_label_zpl2.py [hunk 1]
```diff
@@ -0,0 +1,940 @@
...
+        label = self.new_label()
+        label.print_label(self.printer, self.printer)
+        cups.Connection().printFile.assert_called_once()
+
+    def test_empty_label_contents(self):
+        """ Check contents of an empty label """
+        label = self.new_label()
+        contents = label._generate_zpl2_data(self.printer)
+        self.assertEqual(
+            contents,
```
Comment (gurneyalex) [inline printer_zpl2/tests/test_printing_label_zpl2.py:69]: this is a byte object, and so the comparison with the string is failing.

**Review by gurneyalex — CHANGES_REQUESTED** [review]

**Comment by gurneyalex** [comment]
the tests show other issues with os.write() and string / bytes

File: printer_zpl2/tests/test_printing_label_zpl2.py (hunk 1 above)
Comment (bodedra) [reply printer_zpl2/tests/test_printing_label_zpl2.py:69]: Corresponded. Thank you.

**Comment by Garamotte** [comment]
Hello,

There is already a work in progress for this module : https://github.com/OCA/report-print-send/pull/129
And I have a fixed version of this wip here : https://github.com/syleam/report-print-send/commits/11.0-migrate-printer_zpl2-fmdl

**Comment by bodedra** [comment]
@sylvain-garancher Thank you. I got it. That's my bad to not check issues and PRs. :+1: 

Reference #129

**Pull request closed without merging**